fiovb/ta/entry.c

```diff
@@ -264,7 +264,13 @@ static TEE_Result write_persist_value(uint32_t pt,
 	TEE_MemMove(value, params[1].memref.buffer,
 		    value_sz);
 
-	if (strncmp(name_buf, BOOTFIRM_VER, strlen(BOOTFIRM_VER))) {
+	if (!strncmp(name_buf, vendor_prefix, strlen(vendor_prefix)) &&
+	    !IS_ENABLED(CFG_FIOVB_VENDOR_CREATE)) {
+		res = TEE_ERROR_GENERIC;
```

Comment:
I think it is better to return TEE_ERROR_BAD_PARAMETERS instead, as the issue is associated with the parameter given not matching the built configuration.